### PR TITLE
Port driver manager to ros2 subsystem controller

### PR DESCRIPTION
**Pull request by adev4a**
# PR Details
This PR ports the health monitor's driver functionality to ros2. This port is currently being made to support both ros1 and ros2 drivers. 

## Description
The logic implemented in the driver_subsystem_controller follows the same logic as the ros1 manager, in that it loads the required driver names from carma-config. The ros2 drivers are then managed by the driver_subsystem_controller.
The driver node names specified in carma-config are used define which drivers will be treated as required. 
In order to manage ros1 drivers, updates on driver readiness and errors will still be published on system alert.
System alert will track changes from the required ros2 drivers as well.

## Types of changes

- [ ] Defect fix (non-breaking change that fixes an issue)
- [x] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that cause existing functionality to change)

## Checklist:

- [ ] I have added any new packages to the sonar-scanner.properties file
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING**](https://github.com/usdot-fhwa-stol/carma-platform/blob/develop/Contributing.md) document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

**Review by JonSmet — CHANGES_REQUESTED** [review]

File: subsystem_controllers/src/drivers_controller/drivers_controller_node.cpp [hunk 1]
```diff
@@ -22,6 +22,140 @@ namespace subsystem_controllers
...
+    }
+
+  }
+
+  void DriversControllerNode::timer_callback()
+  {
+    
+    long time_now = this->now().nanoseconds() / 1e6;
+    rclcpp::Duration sd(config_.startup_duration_);
+    long start_duration = sd.nanoseconds()/1e6;
```
[the remark this answers is not in the document]
Comment (adev4a) [reply subsystem_controllers/src/drivers_controller/drivers_controller_node.cpp:117]: They're being used to define start_duration which is passed to the driver_manager

File: subsystem_controllers/include/subsystem_controllers/drivers_controller/driver_manager.hpp [hunk 2]
```diff
@@ -0,0 +1,130 @@
...
+
+        /*!
+         * \brief Check if all critical drivers are operational for truck
+         */
+        std::string are_critical_drivers_operational_truck(long current_time);
+
+        /*!
+         * \brief Check if all critical drivers are operational for car
+         */
+        std::string are_critical_drivers_operational_car(long current_time);
```
[the remark this answers is not in the document]
Comment (adev4a) [reply subsystem_controllers/include/subsystem_controllers/drivers_controller/driver_manager.hpp:79]: Yes, good point

File: subsystem_controllers/src/drivers_controller/drivers_controller_node.cpp (hunk 1 above)
[the remark this answers is not in the document]
Comment (adev4a) [reply subsystem_controllers/src/drivers_controller/drivers_controller_node.cpp:110]: Updated, start duration is now being used to create the driver_manager

**Review by JonSmet — CHANGES_REQUESTED** [review]

File: subsystem_controllers/config/drivers_controller_config.yaml [hunk 3]
```diff
@@ -19,5 +19,21 @@
...
+
+    # List of ros1 camera drivers (node name) to consider required and who's failure shall result in automation abort.
+    ros1_camera_drivers: ['']
+    
     # Boolean: If this flag is true then all nodes under subsystem_namespace are treated as required in addition to any nodes in required_subsystem_nodes
-    full_subsystem_required: false
\ No newline at end of file
+    full_subsystem_required: false
+
+    # Double: The time allocated for system startup in seconds
```
Comment (JonSmet) [inline subsystem_controllers/config/drivers_controller_config.yaml:35]: Let's change this param type to `Integer`

File: subsystem_controllers/include/subsystem_controllers/drivers_controller/driver_manager.hpp [hunk 4]
```diff
@@ -0,0 +1,104 @@
...
+
+    /**
+     * \brief The DriverManager serves as a component to manage CARMA required ROS1 Drivers 
+     */ 
+    class DriverManager
+    {
+        public:
+
+        /*!
+         * \brief Default constructor for DriverManager with driver_timeout_ = 1000
```
Comment (JonSmet) [inline subsystem_controllers/include/subsystem_controllers/drivers_controller/driver_manager.hpp:49]: Let's add units to this comment for the driver_timeout_ setting

**Review by JonSmet — CHANGES_REQUESTED** [review]

**Review by JonSmet — DISMISSED** [review]

**Review by JonSmet — APPROVED** [review]

**Pull request merged**